### PR TITLE
Replace time.After with time.NewTimer

### PR DESCRIPTION
**Pull request by lelenanam**
If the rate of compaction is high we might leave a lot of stray timers.

### CODE REVIEW

File: tsdb/db.go [hunk 1]
```diff
@@ -588,9 +588,9 @@ func (db *DB) run() {
 			return
 		case <-time.After(backoff):
 		}
-
+		timer := time.NewTimer(1 * time.Minute)
```
Comment (codesome) [inline tsdb/db.go]: Ticker makes more sense here. A timer would run the compaction only once.

File: tsdb/db.go [hunk 2]
```diff
@@ -611,8 +611,10 @@ func (db *DB) run() {
 			}
 			db.autoCompactMtx.Unlock()
 		case <-db.stopc:
+			timer.Stop()
```
Comment (codesome) [inline tsdb/db.go]: This could be in a `defer` statement just after initialization.

File: tsdb/db.go (hunk 1 above)
Comment (lelenanam) [reply tsdb/db.go]: Considering backoff logic it isn't that easy to convert per-iteration timer to one ticker.

File: tsdb/db.go (hunk 2 above)
Comment (lelenanam) [reply tsdb/db.go]: defer will hold a reference for each timer in that case and that will be a memory leak.

File: tsdb/db.go (hunk 1 above)
Comment (codesome) [reply tsdb/db.go]: Ah sorry, I didn't notice the loop enclosed this. In that case, the current implementation looks simpler and `time.After` uses `time.NewTimer` internally. 1 object per minute doesn't hurt much.

**Pull request closed without merging**